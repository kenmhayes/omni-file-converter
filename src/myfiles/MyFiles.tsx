import React, { ChangeEvent, useRef } from 'react';
import { Button, Form } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';

function MyFiles() {
  const navigate = useNavigate();
  const sessionId = useRef<string>('');

  const onSessionInputChange = (event: ChangeEvent<HTMLInputElement>): void => {
    const inputValue = event.target.value;
    sessionId.current = inputValue;
  };

  const onSessionButtonClick = (): void => {
    navigate(`/${sessionId.current}`);
  };

  return (
    <div className="container">
      <h3>Find your converted files</h3>
      <Form>
        <Form.Group className="mb-3" controlId="sessionId">
          <Form.Label>Session ID</Form.Label>
          <Form.Control type="text" onChange={onSessionInputChange} />
        </Form.Group>
        <Button type="button" variant="primary" onClick={onSessionButtonClick}>Submit</Button>
      </Form>
    </div>
  );
}

export default MyFiles;
