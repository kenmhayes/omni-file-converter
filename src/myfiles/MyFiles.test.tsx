import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MyFiles from '../myfiles/MyFiles';
import { fireEvent, render, waitFor } from '@testing-library/react';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
}));

it('renders a session search page', () => {
    const component = create(
        <MemoryRouter>
            <MyFiles />
        </MemoryRouter>
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});

it('searches based on session id', async () => {
    const sessionId = '12345';

    const { getByRole, getByLabelText } = render(
        <MemoryRouter>
            <MyFiles />
        </MemoryRouter>
    );

    const input = getByLabelText('Session ID');

    await fireEvent.change(input, { target: { value: sessionId}});

    await fireEvent.click(getByRole('button'));

    expect(mockedUsedNavigate).toHaveBeenCalledWith('/' + sessionId);
});
