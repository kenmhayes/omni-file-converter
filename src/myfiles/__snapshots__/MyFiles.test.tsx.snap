// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a session search page 1`] = `
<div
  className="container"
>
  <h3>
    Find your converted files
  </h3>
  <form
    className=""
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="sessionId"
      >
        Session ID
      </label>
      <input
        className="form-control"
        id="sessionId"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </form>
</div>
`;
